zio-cli/shared/src/main/scala/zio/cli/Options.scala

```diff
@@ -85,18 +94,68 @@ object Options {
         } yield (args -> (a -> b)))
   }
 
-  final case class Requires[A, B](options: Options[A], target: Options[B], predicate: B => Boolean) extends Options[A]
+  final case class Requires[A, B](options: Options[A], target: Options[B], predicate: B => Boolean) extends Options[A] {
+    def validate(args: List[String], opts: ParserOptions): IO[List[HelpDoc.Block], (List[String], A)] = ???
+  }
 
-  final case class RequiresNot[A, B](options: Options[A], target: Options[B], predicate: B => Boolean)
-      extends Options[A]
+  final case class RequiresNot[A, B](options: Options[A], target: Options[B], predicate: B => Boolean) extends Options[A] {
+    def validate(args: List[String], opts: ParserOptions): IO[List[HelpDoc.Block], (List[String], A)] = ???
+  }
 
-  sealed trait Type[+A]
+  sealed trait Type[+A] {
+    def validate(args: List[String], supportedOptions: Vector[String]): IO[List[HelpDoc.Block], (List[String], A)]
+  }
   object Type {
 
-    final case class Toggle(negationName: Option[String], ifPresent: Boolean) extends Type[Boolean]
-    final case class Map[A, B](value: Type[A], f: A => Either[String, B])     extends Type[B]
-    final case class Optional[A](value: Type[A])                              extends Type[Option[A]]
-    final case class Primitive[A](primType: PrimType[A])                      extends Type[A]
+    final case class Toggle(negationName: Option[String], ifPresent: Boolean) extends Type[Boolean] {
```

Comment:
Let's say you have an option with `Toggle(Some("V"), true)`, where the primary name of the `Single` is "v".

Then the interpretation is as follows:

 - If the option is present, then it produces `true`, because `ifPresent = true`. 
 - If the option is absent, then it produces `false`, because `ifPresent = true`.
 - If the option is not present, but the negative name is present (`-V`), then it produces `false`, because `ifPresent` is true.

Does this make sense?

Basically `Toggle` is for binary settings, which have a default, and which maybe have a way to negate them.

Reply:
@jdegoes we can't leave Toggle as type here, because it has more intricacies and thus those implementations can't be fulfilled. It has more custom options traversing like Optional type. If we leave it as type, then we can test for value not being present, since this fires only if value is present indeed. I don't like the idea of duplicating operators for Toggle, but its a bit problematic data type atm.

Reply:
Why don't you leave that one as `Type`, and I can work on fixing it myself? i.e. focus on the other ones for now.

Reply:
Sounds good :wink: